ext/mysqli/mysqli_nonapi.c

```diff
@@ -631,16 +632,23 @@ PHP_FUNCTION(mysqli_query)
 	}
 
 	if (!query_len) {
-		php_error_docref(NULL, E_WARNING, "Empty query");
-		RETURN_FALSE;
+		zend_argument_value_error(ERROR_ARG_POS(2), "cannot be empty");
+		RETURN_THROWS();
 	}
-#ifdef MYSQLI_USE_MYSQLND
-	if ((resultmode & ~MYSQLI_ASYNC) != MYSQLI_USE_RESULT && (resultmode & ~(MYSQLI_ASYNC | MYSQLI_STORE_RESULT_COPY_DATA)) != MYSQLI_STORE_RESULT) {
-#else
-	if ((resultmode & ~MYSQLI_ASYNC) != MYSQLI_USE_RESULT && (resultmode & ~MYSQLI_ASYNC) != MYSQLI_STORE_RESULT) {
-#endif
-		php_error_docref(NULL, E_WARNING, "Invalid value for resultmode");
-		RETURN_FALSE;
+	if ((resultmode & ~MYSQLI_ASYNC) != MYSQLI_USE_RESULT &&
+		MYSQLI_STORE_RESULT !=
+		#ifdef MYSQLI_USE_MYSQLND
```

Comment:
You should be able to just drop the ifdef here. MYSQLI_STORE_RESULT_COPY_DATA is zero if not using mysqlnd.